rocky/rocky/scheduler.py

```diff
@@ -238,9 +255,62 @@ def push_task(self, queue_name: str, prioritized_item: QueuePrioritizedItem) ->
                 raise SchedulerError()
 
     def health(self) -> ServiceHealth:
-        health_endpoint = self.session.get(f"{self._base_uri}/health")
-        health_endpoint.raise_for_status()
-        return ServiceHealth.parse_raw(health_endpoint.content)
+        try:
+            health_endpoint = self.session.get(f"{self._base_uri}/health")
+            health_endpoint.raise_for_status()
+            return ServiceHealth.parse_raw(health_endpoint.content)
+        except ConnectionError:
+            raise SchedulerError()
+
+
+def get_scheduler_client():
+    try:
+        client = SchedulerClient(settings.SCHEDULER_API)
+        client.health()
```

Comment:
This functionality doesn't belong here. If the caller intends to ensure system health, that responsibility should rest with the caller, not within this factory function.

Reply:
When the client is offline, I want to first check if its healthy meaning running and not offline.  I have refactored it to return the client instance anyways but still cannot do anything with it till the Scheduler is back online.

Reply:
> When the client is offline, I want to first check if its healthy meaning do a simple request to the health page. The other thing is LazyTaskList will first do an internal count throwing a ConnectionError inside and I cannot get it out to rocky. My intention was here to even before that when the client is initiated to check if it is usable. I have refactored it to return the client instance anyways but still cannot do anything with it till the Scheduler is back online.

Factory methods, like the one here, shouldn't have side effects such as calling external services.

Performing health checks before using the `LazyTaskList` doesn't help you out, since it doesn't guarantee the service's health state won't change during the lifespan of this `LazyTaskList` instance.

And what do you mean by "cannot get it out to rocky"? I'm sure we can refactor this piece to achieve what we want.